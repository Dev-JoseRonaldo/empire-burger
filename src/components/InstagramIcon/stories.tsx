import { ComponentMeta, ComponentStory } from '@storybook/react'
import Instagram from '.'

export default {
  title: 'Components/Instagram',
  component: Instagram,
  parameters: {
    backgrounds: {
      default: 'dark'
    },
    design: {
      type: 'figma',
      url: 'URL_FIGMA'
    }
  },
  args: {
    iconColor: 'primary'
  }
} as ComponentMeta<typeof Instagram>

const Template: ComponentStory<typeof Instagram> = args => (
  <Instagram {...args} />
)

export const primary = Template.bind({})
primary.args = {}

export const secondary = Template.bind({})
secondary.args = {
  iconColor: 'secondary'
}
