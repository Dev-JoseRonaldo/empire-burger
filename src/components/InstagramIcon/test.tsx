import { render, screen } from '@testing-library/react'
import InstagramIcon from '.'

const { getByLabelText } = screen

describe('<InstagramIcon />', () => {
  it('should render by primary color', () => {
    const { container } = render(<InstagramIcon iconColor="primary" />)

    const instagramIcon = getByLabelText('Instragram icon')
    const path1 = container.querySelector('svg > path')

    expect(instagramIcon).toHaveStyle({
      width: 400,
      height: 80
    })

    expect(path1).toHaveAttribute('fill', '#ffffff')
  })

  it('should render secondary color', () => {
    const { container } = render(
      <InstagramIcon width={400} height={80} iconColor="secondary" />
    )

    const instagramIcon = getByLabelText('Instragram icon')
    const path1 = container.querySelector('svg > path')

    expect(instagramIcon).toHaveAttribute('width', '400')
    expect(instagramIcon).toHaveAttribute('height', '80')

    expect(path1).toHaveAttribute('fill', '#F59A1B')
  })

  it('should match snapshot', () => {
    const { container } = render(<InstagramIcon iconColor="primary" />)

    expect(container).toMatchSnapshot()
  })
})
