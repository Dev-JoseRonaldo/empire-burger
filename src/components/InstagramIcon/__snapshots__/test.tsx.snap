// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InstagramIcon /> should match snapshot 1`] = `
<div>
  <svg
    aria-label="Instragram icon"
    fill="none"
    role="img"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9.502 4.668a4.608 4.608 0 0 0-4.615 4.615 4.608 4.608 0 0 0 4.615 4.615 4.608 4.608 0 0 0 4.615-4.615 4.608 4.608 0 0 0-4.615-4.615Zm0 7.615c-1.65 0-3-1.345-3-3s1.345-3 3-3 3 1.345 3 3-1.35 3-3 3Zm5.88-7.804c0 .599-.482 1.077-1.076 1.077a1.076 1.076 0 1 1 1.076-1.076Zm3.057 1.093c-.069-1.442-.398-2.72-1.454-3.772C15.932.748 14.655.42 13.213.346c-1.486-.084-5.94-.084-7.426 0-1.438.069-2.715.398-3.772 1.45C.96 2.85.634 4.126.561 5.568c-.084 1.486-.084 5.94 0 7.427.069 1.441.398 2.719 1.454 3.771 1.057 1.052 2.33 1.382 3.772 1.454 1.486.084 5.94.084 7.426 0 1.442-.068 2.72-.398 3.772-1.454 1.052-1.052 1.381-2.33 1.454-3.771.084-1.487.084-5.937 0-7.423Zm-1.92 9.017a3.037 3.037 0 0 1-1.711 1.711c-1.185.47-3.997.361-5.306.361-1.31 0-4.125.105-5.306-.361a3.037 3.037 0 0 1-1.71-1.711c-.47-1.185-.362-3.996-.362-5.306s-.105-4.125.361-5.306a3.038 3.038 0 0 1 1.711-1.71c1.185-.47 3.997-.362 5.306-.362 1.31 0 4.125-.105 5.306.361.787.314 1.394.92 1.71 1.711.47 1.185.362 3.997.362 5.306 0 1.31.109 4.125-.361 5.306Z"
      fill="#ffffff"
    />
  </svg>
</div>
`;
